django/contrib/auth/tests/test_remote_user.py

```diff
@@ -145,9 +149,7 @@ def test_user_switch_forces_new_login(self):
 
     def tearDown(self):
         """Restores settings to avoid breaking other tests."""
-        settings.MIDDLEWARE_CLASSES = self.curr_middleware
-        settings.AUTHENTICATION_BACKENDS = self.curr_auth
-
+        pass
```

Comment:
Yes indeed, I should have done:
```
    settings.MIDDLEWARE_CLASSES = self.INITIAL_MIDDLEWARE_CLASSES
```

As it seems, I've been inconsistent in the test files and maybe haven't given enough thought to how the tests are set up and tore down. I'll go through it again and push another commit in about an hour